src/joystick/hidapi/SDL_hidapi_gamecube.c

```diff
@@ -109,10 +109,10 @@ static Uint8 RemapButton(SDL_DriverGameCube_Context *ctx, Uint8 button)
     if (!ctx->m_bUseButtonLabels) {
         /* Use button positions */
         switch (button) {
-        case SDL_GAMEPAD_BUTTON_B:
-            return SDL_GAMEPAD_BUTTON_X;
-        case SDL_GAMEPAD_BUTTON_X:
-            return SDL_GAMEPAD_BUTTON_B;
+        case SDL_GAMEPAD_BUTTON_EAST:
+            return SDL_GAMEPAD_BUTTON_WEST;
+        case SDL_GAMEPAD_BUTTON_WEST:
+            return SDL_GAMEPAD_BUTTON_EAST;
         default:
             break;
         }
```

Comment:
I assume we're modelling the Gamepad controller's unusual layout

```
  (Y)
    [   ] (X)
    [ A ]
(B) [   ]
```

(reference: https://upload.wikimedia.org/wikipedia/commons/thumb/4/41/Gamecube-controller-breakdown.jpg/512px-Gamecube-controller-breakdown.jpg)


as though it was this

```
   (Y)
(B)   (X)
   (A)
```

for the sake of having it be the same shape as the typical Xbox/Playstation/Switch diamond?

If so, shouldn't we now be mapping A = south, B = west, X = east, Y = north unconditionally, regardless of any hints?

Reply:
I would think so... this should be easy to map to NSEW while also binding the primary action to A/South and the secondary action to B/West.

Reply:
Having the Gamecube controller behave as described above is consistent with the convention established by https://github.com/gabomdq/SDL_GameControllerDB for example.

Reply:
I'll have to double check the GameCube mapping. I think we want to treat it like a Nintendo Switch Pro controller and see where the buttons land. I think A (East) is A (big green center button) and B (South) is B (red southwest button), but I'll double check. The N/S/E/Wness of these buttons will be a lie, but I think the behavior for users will be as intended.

Reply:
I have a separate commit coming that removes the button label hint, as the logic is entirely in the new button label and primary/secondary action functions.

Reply:
> The N/S/E/Wness of these buttons will be a lie

Are you modelling the Gamecube's layout as being slanted, with north pointing away from the C-stick, instead of away from the player as it usually does:

```
          _ N
          /|
  (Y)   /
    [   ] (X)
    [   ]
(B) [  A]
```

and adjusting the buttons' logical positions so that N = X, E = A, S = B, W = Y? That would match the Switch layout.

If you're not doing something like that, then it seems to me that having N/S/E/W be a lie would defeat the whole idea of this change - the point of this change is that we want the relative positions of buttons to be what people expect.

Reply:
For what it's worth, https://github.com/gabomdq/SDL_GameControllerDB seems to be treating the Gamecube layout as having "north" slightly (less than 45 degrees) to the left of where it ideally ought to be, so that the A is treated as if it was south, with X, Y and B clustered around it in roughly the same layout as arrow keys or WASD on keyboard; and the fact that there's code to swap X and B in "labels" mode suggests that SDL2 did the same.

If you're now modelling the B as south instead, then your "north" arrow is pointing more than 45 degrees to the right of where I'd expect it to be, which seems odd, but would make the button labelling match Switch controllers.

Reply:
I think that's how the Nintendo Switch GameCube controllers work, e.g.
https://www.amazon.com/PowerA-Wireless-Controller-Nintendo-Switch-GameCube/dp/B07GXLBCC3?th=1

I'll double check though.

Reply:
It would be nice if the new system just worked and we didn't have to know whether or not the Nintendo Switch controller had a GameCube form factor or not, and I can check tomorrow.